package/rotating_zmq_logger/src/rotating_logger.cc

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (C) 2017 Swift Navigation Inc.
+ * Contact: Jonathan Diamond <jonathan@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include "rotating_logger.h"
+
+#include <dirent.h>
+#include <time.h>
+#include <assert.h>
+#include <string.h>
+#include <stdarg.h>
+#include <sys/statvfs.h>
+
+/*
+ * Name formate: xxxx-yyyyy.sbp
+ * First four digit number (xxxx) is the session counter.
+ * This increments each time the process doing the logging is restarted.
+ * The number to start from comes from parsing the files that exist on the SD card.
+ * It starts at 1 rather than 0.
+ * The logger asserts if a session 9999 file already exists.
+ *
+ * The second 5 digit number (yyyyy) is the number of minutes since the linux process
+ * doing the logging started.  It represents the BEGINNING of the logfile.
+ * If minutes exceeds 99999 the session is incremented.
+ */
+static const std::string LOG_SUFFIX = ".sbp";
+static const size_t LOG_NAME_LEN = 4 + 1 + 5 + 4;
+
+static bool debug = true;
+static void debug_printf(const char *msg, ...)
+{
+  if (!debug) {
+    return;
+  }
+
+  va_list ap;
+  va_start(ap, msg);
+  vprintf(msg, ap);
+  va_end(ap);
+}
+
+int RotatingLogger::check_disk_full(void)
+{
+  struct statvfs fs_stats;
+  if (statvfs(_out_dir.c_str(), &fs_stats)) {
+    return -1;
+  }
+  double percent_full = double(fs_stats.f_blocks - fs_stats.f_bavail) / double(fs_stats.f_blocks) * 100.;
+  if (percent_full < _disk_full_threshold) {
+    return 0;
+  } else {
+    return 1;
+  }
+}
+
+bool RotatingLogger::open_new_file()
+{
+  if(_session_count >= 9999) {
+    perror("Max session exceeded");
+    return false;
+  }
+  if (_cur_file != NULL) {
+    fclose(_cur_file);
+    _cur_file = NULL;
+  }
+
+  int fs_status = check_disk_full();
+  if (fs_status  == 1) {
+    debug_printf("Target dir full\n");
```

Comment:
left this as stdout since it's isn't necessarily an error. Communicating this will be more important when we actually enable logging as a setting